model/task/task.go

```diff
@@ -2572,35 +2580,90 @@ func ArchiveMany(tasks []Task) error {
 	defer session.EndSession(ctx)
 
 	txFunc := func(sessCtx mongo.SessionContext) (interface{}, error) {
-		oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
-		_, err = oldTaskColl.InsertMany(ctx, archived)
-		if err != nil {
-			return nil, err
+		var err error
+		if len(toArchive) > 0 {
+			oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
+			_, err = oldTaskColl.InsertMany(sessCtx, toArchive)
+			if err != nil {
+				return nil, errors.Wrap(err, "archiving tasks")
+			}
+		}
+		if len(taskIds) > 0 {
+			_, err = evergreen.GetEnvironment().DB().Collection(Collection).UpdateMany(sessCtx,
+				bson.D{{Key: "_id", Value: bson.D{{Key: "$in", Value: taskIds}}}}, // Query all display tasks and tasks
+				bson.A{ // Pipeline
+					bson.D{{Key: "$set", Value: bson.D{ // execution = execution + 1
```

Comment:
This should just be inc

Reply:
Pipelines don't support the inc operator (at least our version maybe? I test it with the same version as prod and it said unsupported). I did update the updateone call to use inc instead though!

Reply:
This is the old way that this is written. https://github.com/evergreen-ci/evergreen/blob/c8ac8688404d6dd1ac0e420e206238ef5895b98e/model/task/task.go#L2582 You should be able to do something similar.

Reply:
Oh! This is brilliant, let me change from the pipeline version to using this updatemany call. It definitely will be more concise and can use inc

Reply:
I keep getting a memory error, I've been debugging but I am having trouble, could you look over these images when you review (no problem if you can't find the error)
Old:
<img width="775" alt="Screen Shot 2022-08-08 at 3 38 14 PM" src="https://user-images.githubusercontent.com/64446617/183500276-e925ad56-560c-42ea-922d-fa92e973d874.png">
New:
<img width="877" alt="Screen Shot 2022-08-08 at 3 37 22 PM" src="https://user-images.githubusercontent.com/64446617/183500335-b131a821-a3f0-4f27-b1a2-c0a715d47523.png">

Reply:
Updated